### PR TITLE
[#10][part-0]feat(profile): support cpu profiling online 

### PR DESCRIPTION
**Pull request by advancedxy**
After start uniffle server, it should be possible to get cpu pprof via 
```
go tool pprof -http="0.0.0.0:8081" http://localhost:8080/debug/pprof/profile\?seconds\=30\&frequency\=100
```
Note: `localhost:8080` is uniffle's http server.

One profiling result:
image

### CODE REVIEW

**Comment by advancedxy** [comment]
@zuston Please take a look when you have time.

File: src/http/http_service.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            port,
+        }
+    }
+}
+
+impl HTTPServer for WarpServer {
+    fn start(&mut self) {
+        let hello_world = warp::path::end().map(|| "Hello uniffle server");
+
+        // todo: maybe we should replace warp with other rest framework, it's super weired to
```
Comment (zuston) [inline src/http/http_service.rs:32]: Yes. This abstract http service is good for me.

And `warp` crate is hard to wrap different routes. Maybe the `poem` is a better choice. Anyway, let's refactor in the later PRs.

File: src/main.rs [hunk 2]
```diff
@@ -143,8 +136,13 @@ async fn main() -> Result<()> {
     let rpc_port = config.grpc_port.unwrap_or(19999);
     let datanode_uid = gen_datanode_uid(rpc_port);
 
+    let metric_config = config.metrics.clone();
     // start metric service to expose http api
-    start_metric_service(&config.metrics, datanode_uid.clone());
+    let start_http_service = configure_metric_service(&metric_config, datanode_uid.clone());
```
Comment (zuston) [inline src/main.rs:141]: I think this is wrong. The http service is for exposing metrics or triggering dump cpu profile file, or dumping the await tasks trees.

So whether it starts or not has nothing to do with the metric configuration.

**Comment by zuston** [comment]
Overall LGTM. Thanks for your great work!

File: src/http/http_service.rs (hunk 1 above)
Comment (advancedxy) [reply src/http/http_service.rs:32]: Yeah. Maybe we should choose poem or other popular rest framework.

File: src/main.rs (hunk 2 above)
Comment (advancedxy) [reply src/main.rs:141]: I know...

The http service whether be up or not should be determined by its own configuration.

If the metrics configuration is not enabled, then the route for metrics should be omit or disabled. But it's super weird to do conditional routing... Thus I delegate the condition to this one. And it definitely should be refactored once the http framework supports conditional routing.

Comment (zuston) [reply src/main.rs:141]: Got it. Let's merge this firstly

Comment (zuston) [reply src/main.rs:141]: I have refactor this part by using `poem` crate. Please see #39

**Pull request merged**